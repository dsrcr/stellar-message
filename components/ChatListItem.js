import { ListItem, Avatar } from '@rneui/themed';
import tw from 'twrnc';

export default function ChatListItem({ id, chatName, enterChat, chatImage }) {
  return (
    <ListItem key={id} bottomDivider onPress={() => enterChat(id, chatName)}>
      <Avatar
        rounded
        source={{
          uri: 'https://exhibitdev.files.wordpress.com/2014/01/bch8ogfcyaadpji-jpg-large.jpeg',
        }}
      />
      <ListItem.Content>
        <ListItem.Title style={{ fontWeight: '800' }}>{chatName}</ListItem.Title>
        <ListItem.Subtitle numberOfLines={1} ellipsizeMode="tail">
          This is a test subtitle
        </ListItem.Subtitle>
      </ListItem.Content>
    </ListItem>
  );
}
