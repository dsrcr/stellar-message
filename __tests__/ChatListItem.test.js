import renderer from 'react-test-renderer';
import ChatListItem from '../components/ChatListItem';

describe('ChatListItem', () => {
  const mockEnterChat = jest.fn();

  const testData = {
    id: '1',
    chatName: 'Test Chat',
  };

  test('renders ChatListItem correctly', () => {
    const tree = renderer.create(<ChatListItem {...testData} enterChat={mockEnterChat} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
