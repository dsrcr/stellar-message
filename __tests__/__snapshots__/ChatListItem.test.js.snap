// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatListItem renders ChatListItem correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomWidth": 0.5,
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
    testID="RNE__LISTITEM__padView"
  >
    <View
      style={
        {
          "borderRadius": 17,
          "display": "flex",
          "flexDirection": "column",
          "height": 34,
          "justifyContent": "center",
          "width": 34,
        }
      }
      theme={
        {
          "colors": {
            "background": "#ffffff",
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": {
              "android": {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ad1457",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
          "spacing": {
            "lg": 12,
            "md": 8,
            "sm": 4,
            "xl": 24,
            "xs": 2,
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 17,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={17}
          onLoad={[Function]}
          source={
            {
              "uri": "https://exhibitdev.files.wordpress.com/2014/01/bch8ogfcyaadpji-jpg-large.jpeg",
            }
          }
          style={
            {
              "bottom": 0,
              "flex": 1,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="RNE__Avatar__Image"
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={true}
          collapsable={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
        <View
          style={
            {
              "flex": 1,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        {
          "paddingLeft": 16,
        }
      }
    />
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      theme={
        {
          "colors": {
            "background": "#ffffff",
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": {
              "android": {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ad1457",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
          "spacing": {
            "lg": 12,
            "md": 8,
            "sm": 4,
            "xl": 24,
            "xs": 2,
          },
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "#242424",
            "fontSize": 17,
            "fontWeight": "800",
          }
        }
        testID="listItemTitle"
      >
        Test Chat
      </Text>
      <Text
        accessibilityRole="text"
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "backgroundColor": "transparent",
            "color": "#242424",
            "fontSize": 15,
          }
        }
        testID="listItemTitle"
      >
        This is a test subtitle
      </Text>
    </View>
  </View>
</View>
`;
